### ISSUE TITLE
Segmentation fault on startup in sqlcipher at version 4.2.0

### ISSUE THREAD

**Issue by anthonybilinski** [qTox/qTox#6077]
Taken from https://github.com/qTox/qTox/issues/4054#issuecomment-546622526

##### Brief Description

 OS: Slackware Current 32bit
qTox: 1.17-dev (on this time)
toxcore: 0.2.10
Qt: 5.13.1

##### Steps to reproduce

1. Start qtox
2. Login to created profile

##### Observed Behavior

Segfault on start.

##### Additional Info
```
(gdb) run
Starting program: /usr/bin/qtox 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/libthread_db.so.1".
[New Thread 0xae687b40 (LWP 10432)]
[14:43:25.256 UTC] :0 : Warning: QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-igor'
[New Thread 0xadb8db40 (LWP 10433)]
[14:43:25.281 UTC] persistence/settings.cpp:120 : Debug: No settings file found, using defaults
[14:43:25.281 UTC] persistence/settings.cpp:124 : Debug: "Loading settings from :/conf/qtox.ini"
[14:43:25.289 UTC] ipc.cpp:61 : Debug: Our global IPC ID is  12848374570953043189
[14:43:25.289 UTC] main.cpp:308 : Debug: commit:  build without git
[New Thread 0xad1ffb40 (LWP 10434)]
[New Thread 0xa886cb40 (LWP 10436)]
[New Thread 0xa7dbeb40 (LWP 10437)]
[14:43:32.306 UTC] persistence/profile.cpp:146 : Debug: Loading tox save  "/home/igor/.config/tox/Igor.tox"
[14:43:32.359 UTC] persistence/settings.cpp:494 : Debug: Loading personal settings from "/home/igor/.config/tox/Igor.ini"
[14:43:32.360 UTC] core/toxoptions.cpp:110 : Debug: Core starting with IPv6 enabled
[New Thread 0xa7273b40 (LWP 10439)]
[New Thread 0xa68ffb40 (LWP 10440)]
[14:43:32.418 UTC] persistence/db/rawdatabase.cpp:285 : Debug: Setting SQLCipher 4.x parameters

Thread 8 "qTox Database" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0xa68ffb40 (LWP 10440)]
0xb40a04eb in EVP_MD_CTX_clear_flags () from /usr/lib/libcrypto.so.1.1
(gdb) bt
#0  0xb40a04eb in EVP_MD_CTX_clear_flags () from /usr/lib/libcrypto.so.1.1
#1  0xb4090926 in EVP_DigestInit_ex () from /usr/lib/libcrypto.so.1.1
#2  0xb40aa13d in HMAC_Init_ex () from /usr/lib/libcrypto.so.1.1
#3  0xb5a716c7 in ?? () from /usr/lib/libsqlcipher.so.0
#4  0xb5a7e99c in ?? () from /usr/lib/libsqlcipher.so.0
#5  0xb5a867ce in sqlcipher_page_cipher () from /usr/lib/libsqlcipher.so.0
#6  0xb5a95c5f in ?? () from /usr/lib/libsqlcipher.so.0
#7  0xb5aa15f9 in ?? () from /usr/lib/libsqlcipher.so.0
#8  0xb5ab18a5 in ?? () from /usr/lib/libsqlcipher.so.0
#9  0xb5a6b4e8 in ?? () from /usr/lib/libsqlcipher.so.0
#10 0xb5ab3170 in ?? () from /usr/lib/libsqlcipher.so.0
#11 0xb5af1f01 in ?? () from /usr/lib/libsqlcipher.so.0
#12 0xb5af2072 in ?? () from /usr/lib/libsqlcipher.so.0
#13 0xb5af20b4 in ?? () from /usr/lib/libsqlcipher.so.0
#14 0xb5af2733 in ?? () from /usr/lib/libsqlcipher.so.0
#15 0xb5af2a07 in ?? () from /usr/lib/libsqlcipher.so.0
#16 0xb5af2e2c in ?? () from /usr/lib/libsqlcipher.so.0
#17 0xb5a798d7 in ?? () from /usr/lib/libsqlcipher.so.0
#18 0xb5a7b065 in ?? () from /usr/lib/libsqlcipher.so.0
#19 0xb5ac39f8 in ?? () from /usr/lib/libsqlcipher.so.0
#20 0xb5ae11c3 in ?? () from /usr/lib/libsqlcipher.so.0
#21 0xb5ae5884 in ?? () from /usr/lib/libsqlcipher.so.0
#22 0xb5ae5ae4 in ?? () from /usr/lib/libsqlcipher.so.0
#23 0x0832b9d5 in RawDatabase::process() ()
#24 0x08268ec2 in ?? ()
#25 0xb4ae567f in QMetaMethod::invoke(QObject*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) const () from /usr/lib/libQt5Core.so.5
#26 0xb4ae7849 in QMetaObject::invokeMethod(QObject*, char const*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) () from /usr/lib/libQt5Core.so.5
#27 0x0832cf65 in QMetaObject::invokeMethod(QObject*, char const*, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) ()
#28 0x08329cf2 in RawDatabase::execNow(QVector<RawDatabase::Query> const&) ()
#29 0x08329a91 in RawDatabase::execNow(RawDatabase::Query const&) ()
#30 0x083299cd in RawDatabase::execNow(QString const&) ()
#31 0x08328e25 in RawDatabase::testUsable() ()
#32 0x08328c1a in RawDatabase::openEncryptedDatabaseAtLatestVersion(QString const&) ()
#33 0x08328b51 in RawDatabase::open(QString const&, QString const&) ()
#34 0x08268e18 in ?? ()
#35 0xb4b03b33 in QMetaCallEvent::placeMetaCall(QObject*) () from /usr/lib/libQt5Core.so.5
#36 0xb4b07903 in QObject::event(QEvent*) () from /usr/lib/libQt5Core.so.5
#37 0xb77415f8 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /usr/lib/libQt5Widgets.so.5
#38 0xb774bb46 in QApplication::notify(QObject*, QEvent*) () from /usr/lib/libQt5Widgets.so.5
#39 0xb4ad5a0d in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /usr/lib/libQt5Core.so.5
#40 0xb4ad5c6b in QCoreApplication::sendEvent(QObject*, QEvent*) () from /usr/lib/libQt5Core.so.5
#41 0xb4ad8cf9 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () from /usr/lib/libQt5Core.so.5
#42 0xb4ad9253 in QCoreApplication::sendPostedEvents(QObject*, int) () from /usr/lib/libQt5Core.so.5
#43 0xb4b387c2 in ?? () from /usr/lib/libQt5Core.so.5
#44 0xb0d8b7ae in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
#45 0xb0d8ba48 in ?? () from /usr/lib/libglib-2.0.so.0
#46 0xb0d8bae5 in g_main_context_iteration () from /usr/lib/libglib-2.0.so.0
#47 0xb4b37ba4 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQt5Core.so.5
#48 0xb4ad4032 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /usr/lib/libQt5Core.so.5
#49 0xb48d7bee in QThread::exec() () from /usr/lib/libQt5Core.so.5
#50 0xb48d7cab in QThread::run() () from /usr/lib/libQt5Core.so.5
#51 0xb48d8ffc in ?? () from /usr/lib/libQt5Core.so.5
#52 0xb42fa517 in start_thread () from /lib/libpthread.so.0
#53 0xb442a146 in clone () from /lib/libc.so.6
```

Thanks to @isav83 for the report.

**Comment by isav83** [comment]
I migrated to 64-bit Slackware and error not reproducing with sqlcipher 4.2.0. qTox works without crash.
**Brief Description:**
OS: Slackware Current 64bit
qTox: 1.17-dev (26 March)
toxcore: 0.2.11
Qt: 5.13.2

**Comment by anthonybilinski** [comment]
I think it's likely it was a packaging or environment issue in that case, or possibly something that was fixed in qTox since the v1.16.3 release (though I'm not aware of something that went it that would fix what you saw). I'll close the issue without further investigation. Please comment if you see it again, and thanks for following up with more info :)
